hearth/lib/hearth/plugin_list.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module Hearth
+  # A list of plugins that can be applied to config. Plugins are callables
+  # that take one argument (config) and are called during client initialization
+  # or operation invocation to modify config.
+  class PluginList
+    include Enumerable
+
+    # Initialize a PluginList.
+    #
+    # @param [Array] ([]) plugins A list of plugins to initialize with
+    def initialize(plugins = [])
+      unless plugins.respond_to?(:each)
```

Comment:
I think you can just check is_a?(Enumerable) right? I don't know if just writing :each is common, but I know it's a requirement for Enumerable anyway.

Reply:
I'm not sure if writing just an `each` method is common or not - but its all that is required for this code to work, so I went with the smallest interface - we don't really need it to be Enumerable.

Reply:
Each is fine, I guess enumerable gives other methods that may not be desirable.